parser-typechecker/src/Unison/Codebase/Type.hs

```diff
@@ -164,7 +166,16 @@ data Codebase m v a = Codebase
     -- `beforeImpl b1 b2` is undefined if `b2` not in the codebase
     --
     --  Use `Codebase.before` which wraps this in a nice API.
-    beforeImpl :: Maybe (Branch.Hash -> Branch.Hash -> m Bool)
+    beforeImpl :: Maybe (Branch.Hash -> Branch.Hash -> m Bool),
+    -- Use the name lookup index to build a 'Names' for all names found within 'Path' of the current root namespace.
+    -- or within the whole root namespace if provided Nothing.
+    --
+    -- NOTE: this method requires an up-to-date name lookup index, which is
```

Comment:
Oh, the name lookup index isn't kept up-to-date automatically? How's it kept up-to-date, then?

Reply:
The current plan is for Enlil to manually call the update function on the codebase during a root-namespace push.
I'd like to eventually get the index fast enough to use it in UCM, but ATM it would add ~1-1.5s to every single operation which affects the root namespace. There are definitely faster ways to keep it up to date than just rebuilding the whole index at once, but for now we'll just use it in Enlil.